lib/ansible/modules/crypto/openssl_csr.py

```diff
@@ -497,29 +585,345 @@ def _check_signature(csr):
             except crypto.Error:
                 return False
 
-        if not state_and_perms:
-            return False
-
         csr = crypto_utils.load_certificate_request(self.path)
 
         return _check_subject(csr) and _check_extensions(csr) and _check_signature(csr)
 
-    def dump(self):
-        '''Serialize the object into a dictionary.'''
 
-        result = {
-            'privatekey': self.privatekey_path,
-            'filename': self.path,
-            'subject': self.subject,
-            'subjectAltName': self.subjectAltName,
-            'keyUsage': self.keyUsage,
-            'extendedKeyUsage': self.extendedKeyUsage,
-            'basicConstraints': self.basicConstraints,
-            'ocspMustStaple': self.ocspMustStaple,
-            'changed': self.changed
-        }
+class CertificateSigningRequestCryptography(CertificateSigningRequestBase):
 
-        return result
+    def __init__(self, module):
+        super(CertificateSigningRequestCryptography, self).__init__(module)
+        self.cryptography_backend = cryptography.hazmat.backends.default_backend()
+
+    def _get_name_oid(self, id):
+        if id in ('CN', 'commonName'):
+            return cryptography.x509.oid.NameOID.COMMON_NAME
+        if id in ('C', 'countryName'):
```

Comment:
I decided to use the current pattern for two reasons:
1. Both are equivalent, and it is easy enough to see that (since here we have `if condition: return value` all the time);
2. The current version is shorter (no `el` in every second line) and the first comparison is not different from all other ones.
I would have rather used a `switch`-like statement for this, but Python doesn't have this...

Anyway, I don't mind converting it if you prefer that.

Reply:
I don't know if `if ... elif ... else` leads to more efficient code being generated or not, then again (unfortunately) ansible modules are not written for execution speed in general.

I have no strong opinion either way, generally I tend to use the `elif` style when needing something like a switch statement in Python (Stackoverflow seems to prefer dictionaries: https://stackoverflow.com/questions/60208/replacements-for-switch-statement-in-python). Since this likely will need to be factored out in the future to the module_utils file, it can be tackled later too. It is obviously correct in the current form, so I'd say just leave it for now.

Reply:
Using a `dict` is also a good idea, but then one has to take care to avoid problems when `cryptography` is not around or some constants require a new enough `cryptography` version...